packages/utils/src/lib/collect/index.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  unrefinedCoreConfigSchema,
+  CoreConfigSchema,
+  globalCliArgsSchema,
+  GlobalCliArgsSchema,
+  refineCoreConfig,
+  runnerOutputSchema,
+} from '@quality-metrics/models';
+import { executePlugins } from './implementation/execute-plugin';
+import { z } from 'zod';
+
+/**
+ * Define Zod schema for the RunAndCollectOptions type
+ *
+ * @example
+ *
+ * // Example data
+ * const raw = {
+ *   ...
+ * };
+ *
+ * // Validate the data against the schema
+ * const validationResult = runAndCollectOptions.safeParse(raw);
+ *
+ * if (validationResult.success) {
+ *   console.log('Valid config:', validationResult.data);
+ * } else {
+ *   console.error('Invalid config:', validationResult.error);
+ * }
+ *
+ */
+export const runAndCollectOptions = refineCoreConfig(
+  z
+    .object({
+      parallel: z
+        .preprocess(
+          a => parseInt(z.string().parse(a.toString()), 10),
+          z.number().positive(),
+        )
+        .optional()
+        .default(1),
+    })
+    .merge(globalCliArgsSchema)
+    .merge(unrefinedCoreConfigSchema),
+);
+
+export type RunAndCollectOptions = z.infer<typeof runAndCollectOptions> &
+  GlobalCliArgsSchema &
+  CoreConfigSchema;
+
+/**
+ * Define Zod schema for the RunAndCollectOutputSchema type
+ *
+ * @example
+ *
+ * // Example data
+ * const raw = {
+ *   ...
+ * };
+ *
+ * // Validate the data against the schema
+ * const validationResult = runAndCollectOutputSchema.safeParse(raw);
+ *
+ * if (validationResult.success) {
+ *   console.log('Valid config:', validationResult.data);
+ * } else {
+ *   console.error('Invalid config:', validationResult.error);
+ * }
+ *
+ */
+export const runAndCollectOutputSchema = runnerOutputSchema.merge(
+  z.object(
+    {
+      date: z.string({ description: 'Date of the run' }),
```

Comment:
It wasn't clear to me until I read this implementation that the `date` is when it finished, not when it started. It would be helpful to specify this in the description:

```suggestion
      date: z.string({ description: 'Date and time when the run finished' }),
```

Reply:
I made it the start date. this is more align with other time series data like a e.g. network request

Reply:
Agreed, start date is more intuitive for me also :+1: